iree/base/BUILD

```diff
@@ -392,33 +365,20 @@ cc_test(
     ],
 )
 
-cc_library(
-    name = "source_location",
-    hdrs = ["source_location.h"],
-    deps = platform_trampoline_deps("source_location"),
-)
-
 cc_library(
     name = "status",
     hdrs = ["status.h"],
     deps = [
-        ":source_location",
-    ] + platform_trampoline_deps("status"),
-)
-
-cc_library(
-    name = "status_matchers",
-    testonly = True,
-    hdrs = ["status_matchers.h"],
-    deps = platform_trampoline_deps("status_matchers"),
+        "//iree/base/internal:status_internal",
```

Comment:
My goal is to remove all of the google/ deps, at which point we'll just have internal/ (matching absl/base/internal/). As more of this style of cleanup happens the goal is to move internal things not based on the C API (such as DynamicLibrary, flatbuffer_util, etc) into internal/ such that only files in base/ are public visibility.